File: src/main/java/org/yggdrasil/core/ledger/transaction/TransactionInPoint.java
package org.yggdrasil.core.ledger.transaction;

import java.io.Serializable;
import java.math.BigInteger;


public class TransactionInPoint implements Serializable {

    protected final Transaction txn;
    protected final BigInteger value;

    public TransactionInPoint() {
        this.txn = null;
        this.value = BigInteger.valueOf(-1);
    }

    public TransactionInPoint(Transaction txn, BigInteger value) {
        this.txn = txn;
        this.value = value;
    }

    public boolean isNull() {
        return (this.txn == null && this.value.compareTo(BigInteger.valueOf(-1)) == 0);
    }

}

File: yggdrasil-core-csharp/yggdrasil-core/core/ledger/transaction/TransactionInPoint.cs
namespace yggdrasil_core.core.ledger.transaction
{
    public class TransactionInPoint : LedgerHashableItem
    {
        private readonly Transaction _txn;
        public Transaction Txn { get { return _txn; } }
        private readonly Coin _value;
        public Coin Value { get { return _value; } }

        public TransactionInPoint(Transaction txn)
        {
            _txn = txn;
            _value = txn.Value();
        }

        public bool IsNull()
        {
            return (_txn == null && _value == null);
        }

        public byte[] Bytes()
        {
            byte[] bytes = new byte[0];
            bytes = (byte[])bytes.Concat(_txn.Bytes());
            bytes = (byte[])bytes.Concat(_value.Bytes());
            return bytes;
        }
    }
}
